core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -258,6 +245,8 @@ private[ui] class StagePage(parent: StagesTab, store: AppStatusStore) extends We
 
     if (!TIMELINE_ENABLED) return Seq.empty[Node]
 
+    val tasks = tasksFunc()
```

Comment:
Thank you. It's reasonable to avoid the calsulation if the timeline feature is disabled.